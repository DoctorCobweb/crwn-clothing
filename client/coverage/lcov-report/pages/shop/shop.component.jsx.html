<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for pages/shop/shop.component.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">pages/shop</a> shop.component.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">72.73% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>8/11</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">87.5% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>7/8</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet"><a name='L1'></a><a href='#L1'>1</a>
<a name='L2'></a><a href='#L2'>2</a>
<a name='L3'></a><a href='#L3'>3</a>
<a name='L4'></a><a href='#L4'>4</a>
<a name='L5'></a><a href='#L5'>5</a>
<a name='L6'></a><a href='#L6'>6</a>
<a name='L7'></a><a href='#L7'>7</a>
<a name='L8'></a><a href='#L8'>8</a>
<a name='L9'></a><a href='#L9'>9</a>
<a name='L10'></a><a href='#L10'>10</a>
<a name='L11'></a><a href='#L11'>11</a>
<a name='L12'></a><a href='#L12'>12</a>
<a name='L13'></a><a href='#L13'>13</a>
<a name='L14'></a><a href='#L14'>14</a>
<a name='L15'></a><a href='#L15'>15</a>
<a name='L16'></a><a href='#L16'>16</a>
<a name='L17'></a><a href='#L17'>17</a>
<a name='L18'></a><a href='#L18'>18</a>
<a name='L19'></a><a href='#L19'>19</a>
<a name='L20'></a><a href='#L20'>20</a>
<a name='L21'></a><a href='#L21'>21</a>
<a name='L22'></a><a href='#L22'>22</a>
<a name='L23'></a><a href='#L23'>23</a>
<a name='L24'></a><a href='#L24'>24</a>
<a name='L25'></a><a href='#L25'>25</a>
<a name='L26'></a><a href='#L26'>26</a>
<a name='L27'></a><a href='#L27'>27</a>
<a name='L28'></a><a href='#L28'>28</a>
<a name='L29'></a><a href='#L29'>29</a>
<a name='L30'></a><a href='#L30'>30</a>
<a name='L31'></a><a href='#L31'>31</a>
<a name='L32'></a><a href='#L32'>32</a>
<a name='L33'></a><a href='#L33'>33</a>
<a name='L34'></a><a href='#L34'>34</a>
<a name='L35'></a><a href='#L35'>35</a>
<a name='L36'></a><a href='#L36'>36</a>
<a name='L37'></a><a href='#L37'>37</a>
<a name='L38'></a><a href='#L38'>38</a>
<a name='L39'></a><a href='#L39'>39</a>
<a name='L40'></a><a href='#L40'>40</a>
<a name='L41'></a><a href='#L41'>41</a>
<a name='L42'></a><a href='#L42'>42</a>
<a name='L43'></a><a href='#L43'>43</a>
<a name='L44'></a><a href='#L44'>44</a>
<a name='L45'></a><a href='#L45'>45</a>
<a name='L46'></a><a href='#L46'>46</a>
<a name='L47'></a><a href='#L47'>47</a>
<a name='L48'></a><a href='#L48'>48</a>
<a name='L49'></a><a href='#L49'>49</a>
<a name='L50'></a><a href='#L50'>50</a>
<a name='L51'></a><a href='#L51'>51</a>
<a name='L52'></a><a href='#L52'>52</a>
<a name='L53'></a><a href='#L53'>53</a>
<a name='L54'></a><a href='#L54'>54</a>
<a name='L55'></a><a href='#L55'>55</a>
<a name='L56'></a><a href='#L56'>56</a>
<a name='L57'></a><a href='#L57'>57</a>
<a name='L58'></a><a href='#L58'>58</a>
<a name='L59'></a><a href='#L59'>59</a>
<a name='L60'></a><a href='#L60'>60</a>
<a name='L61'></a><a href='#L61'>61</a>
<a name='L62'></a><a href='#L62'>62</a>
<a name='L63'></a><a href='#L63'>63</a>
<a name='L64'></a><a href='#L64'>64</a>
<a name='L65'></a><a href='#L65'>65</a>
<a name='L66'></a><a href='#L66'>66</a>
<a name='L67'></a><a href='#L67'>67</a>
<a name='L68'></a><a href='#L68'>68</a>
<a name='L69'></a><a href='#L69'>69</a>
<a name='L70'></a><a href='#L70'>70</a>
<a name='L71'></a><a href='#L71'>71</a>
<a name='L72'></a><a href='#L72'>72</a>
<a name='L73'></a><a href='#L73'>73</a></td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { useEffect, lazy, Suspense } from 'react';
import { Route } from 'react-router-dom';
import { connect } from 'react-redux';
&nbsp;
import Spinner from '../../components/spinner/spinner.component';
&nbsp;
import { fetchCollectionsStart } from '../../redux/shop/shop.actions';
&nbsp;
import { ShopPageContainer } from './shop.styles';
&nbsp;
const CollectionsOverviewContainer = lazy(() =&gt; import('../../components/collections-overview/collections-overview.container'));
const CollectionPageContainer = lazy(<span class="fstat-no" title="function not covered" >()</span> =&gt; <span class="cstat-no" title="statement not covered" >import('../collection/collection.container'))</span>;
&nbsp;
&nbsp;
export const ShopPage = ({ match, fetchCollectionsStart }) =&gt; {
  // IMPORTANT CAVEAT: how to make useEffect act like componentDidMount class life-cycle method
  //
  // usually to mimick onComponentDidMount we pass an [] into useEffect.
  // it means useEffect will only be called once.
  //
  // but doing so will cause react to warn about needing to pass in something
  // to useEffect. also, if we do just use [] then useEffect will be triggered
  // everytime 
  // i) ShopPage rerenders
  // ii) props to ShopPage change
  // iii) if ShopPage had state via useState and state changed
  // iv) if ShopPage's parent rerenders
  //
  // when the app loads and we're on shop page we call fetchCollectionsStart() in useEffect
  // then our app checks for user login. if user is then it sets
  // currentUser in redux to be the user.
  //
  // since app component has currentUser as a prop, the change in currentUser causes app
  // component to re-render, which will cause ShopPage to re-render, which
  // causes the useEffect hook to fire again, which calls fetchCollectionsStart.
  // (try setting useEffect to have [] as second param, reload app and watch
  // the collections in shop page 'flicker' ie. load and then wipe screen to white, then load again)
  //
  // the following is a little hacky but circumvents this flickering behaviour:
  // put the action method in [], as below.
  //
  // so useEffect will fire on ShopPage component load, but only list for changes in
  // fetchCollectionsStart. but fetchCollectionsStart is a method which doesn't
  // change! so if parent App component rerenders due to signing a user in,
  // the useEffect hook wont re-fire..in.because it's only 'listening' to changes in
  // the value of the fetchCollectionsStart action, which by its very nature of just
  // being a 'hard coded' function, won't change.k
  useEffect(() =&gt; {
    fetchCollectionsStart();
  }, [fetchCollectionsStart])
&nbsp;
  return (
    &lt;ShopPageContainer&gt;
      &lt;Suspense fallback={&lt;Spinner /&gt;}&gt;
        &lt;Route
          exact
          path={`${match.path}`}
          component={CollectionsOverviewContainer}
        /&gt;
        &lt;Route
          path={`${match.path}/:collectionId`}
          component={CollectionPageContainer}
        /&gt;
      &lt;/Suspense&gt;
    &lt;/ShopPageContainer&gt;
  );
}
&nbsp;
const mapDispatchToProps = <span class="fstat-no" title="function not covered" >di</span>spatch =&gt; (<span class="cstat-no" title="statement not covered" >{</span>
  fetchCollectionsStart: <span class="fstat-no" title="function not covered" >()</span> =&gt; <span class="cstat-no" title="statement not covered" >dispatch(fetchCollectionsStart()),</span>
});
&nbsp;
export default connect(null, mapDispatchToProps)(ShopPage);</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Aug 06 2019 16:03:04 GMT+1000 (Australian Eastern Standard Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
<script src="../../block-navigation.js"></script>
</body>
</html>
