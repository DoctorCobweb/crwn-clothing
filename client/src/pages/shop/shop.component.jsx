import React, { useEffect, lazy, Suspense } from 'react';
import { Route } from 'react-router-dom';
import { connect } from 'react-redux';

// import CollectionsOverviewContainer from '../../components/collections-overview/collections-overview.container';
// import CollectionPageContainer from '../collection/collection.container';

import Spinner from '../../components/spinner/spinner.component';

import {
  fetchCollectionsStart,
} from '../../redux/shop/shop.actions';

const CollectionsOverviewContainer = lazy(() => import('../../components/collections-overview/collections-overview.container'));
const CollectionPageContainer = lazy(() => import('../collection/collection.container'));


const ShopPage = ({ match, fetchCollectionsStart }) => {
  
  // IMPORTANT CAVEAT: how to make useEffect act like componentDidMount class life-cycle method
  //
  // usually to mimick onComponentDidMount we pass an [] into useEffect.
  // it means useEffect will only be called once.
  //
  // but doing so will cause react to warn about needing to pass in something
  // to useEffect. also, if we do just use [] then useEffect will be triggered
  // everytime 
  // i) ShopPage rerenders
  // ii) props to ShopPage change
  // iii) if ShopPage had state via useState and state changed
  // iv) if ShopPage's parent rerenders
  //
  // when the app loads and we're on shop page we call fetchCollectionsStart() in useEffect
  // then our app checks for user login. if user is then it sets
  // currentUser in redux to be the user.
  //
  // since app component has currentUser as a prop, the change in currentUser causes app
  // component to re-render, which will cause ShopPage to re-render, which
  // causes the useEffect hook to fire again, which calls fetchCollectionsStart.
  // (try setting useEffect to have [] as second param, reload app and watch
  // the collections in shop page 'flicker' ie. load and then wipe screen to white, then load again)
  //
  // the following is a little hacky but circumvents this flickering behaviour:
  // put the action method in [], as below.
  //
  // so useEffect will fire on ShopPage component load, but only list for changes in
  // fetchCollectionsStart. but fetchCollectionsStart is a method which doesn't
  // change! so if parent App component rerenders due to signing a user in,
  // the useEffect hook wont re-fire..in.because it's only 'listening' to changes in
  // the value of the fetchCollectionsStart action, which by its very nature of just
  // being a 'hard coded' function, won't change.k
  useEffect(() => {
    fetchCollectionsStart();
  }, [fetchCollectionsStart])

  return (
    <div className='shop-page'>
      <Suspense fallback={<Spinner />}>
        <Route
          exact
          path={`${match.path}`}
          component={CollectionsOverviewContainer}
        />
        <Route
          path={`${match.path}/:collectionId`}
          component={CollectionPageContainer}
        />
      </Suspense>

    </div>
  );
}

const mapDispatchToProps = dispatch => ({
  fetchCollectionsStart: () => dispatch(fetchCollectionsStart()),
});

export default connect(null, mapDispatchToProps)(ShopPage);