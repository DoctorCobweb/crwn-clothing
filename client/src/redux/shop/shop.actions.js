import { firestore, convertCollectionsSnapshotToMap } from '../../firebase/firebase.utils';


import ShopActionTypes from './shop.types';

export const fetchCollectionsStart = () => ({
  type: ShopActionTypes.FETCH_COLLECTIONS_START,
});

export const fetchCollectionsSuccess = collectionsMap => ({
  type: ShopActionTypes.FETCH_COLLECTIONS_SUCCESS,
  payload: collectionsMap
});

export const fetchCollectionsFailure = errorMessage => ({
  type: ShopActionTypes.FETCH_COLLECTIONS_FAILURE,
  payload: errorMessage,
});


// thunk: action creator that returns a function that gets the dispatch
//
// NOTE: i don't think this is used anymore since we've switched over to sagas
// to handle async actions
export const fetchCollectionsStartAsync = () => {
  return (dispatch) => {
    const collectionRef = firestore.collection('collections');

    // synchronous code
    dispatch(fetchCollectionsStart());

    // asynchronous code
    // using promises to get data
    collectionRef.get().then(snapshot => {
      const collectionsMap = convertCollectionsSnapshotToMap(snapshot);
      dispatch(fetchCollectionsSuccess(collectionsMap));
    }).catch(error => {
      dispatch(fetchCollectionsFailure(error.message))
    });
  };
};

