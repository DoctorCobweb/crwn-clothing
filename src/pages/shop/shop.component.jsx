import React from 'react';
import { Route } from 'react-router-dom';
import { connect } from 'react-redux';

import CollectionsOverviewContainer from '../../components/collections-overview/collections-overview.container';
import CollectionPageContainer from '../collection/collection.container';

import {
  updateCollections,
  fetchCollectionsStartAsync,
} from '../../redux/shop/shop.actions';

import { firestore, convertCollectionsSnapshotToMap } from '../../firebase/firebase.utils';


class ShopPage extends React.Component {
  // needed for the Observer pattern
  // state = {
  //   loading: true,
  // };
  // unsubscribeFromSnapshot = null;

  componentDidMount() {
    console.log('componentWill mount shop.component method');
    const { fetchCollectionsStartAsync } = this.props;
    fetchCollectionsStartAsync();


    // using the Observer pattern
    // const { updateCollections } = this.props;
    // const collectionRef = firestore.collection('collections');
    // this.unsubscribeFromSnapshot = collectionRef.onSnapshot(async snapshot => {
    //   const collectionsMap = convertCollectionsSnapshotToMap(snapshot);

    //   // console.log('asdf');
    //   // console.log(snapshot);
    //   // console.log('collectionsMap');
    //   // console.log(collectionsMap);

    //   updateCollections(collectionsMap);
    //   this.setState({loading: false});
    // });
  }

  componentWillUnmount() {
     // for the Observer pattern
    // this.unsubscribeFromSnapshot();
  }

  render () {
    const { match } = this.props;
    console.log('render shop.component method');

    return (
      <div className='shop-page'>
        <Route
          exact
          path={`${match.path}`}
          component={CollectionsOverviewContainer}
        />

        <Route
          path={`${match.path}/:collectionId`}
          component={CollectionPageContainer}
        />
      </div>
    );
  }
}


const mapDispatchToProps = dispatch => ({
  // updateCollections: collectionsMap => dispatch(updateCollections(collectionsMap)),
  fetchCollectionsStartAsync: () => dispatch(fetchCollectionsStartAsync()),
});

export default connect(null, mapDispatchToProps)(ShopPage);