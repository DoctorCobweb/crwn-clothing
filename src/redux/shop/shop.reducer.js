import ShopActionTypes from './shop.types';

const INITIAL_STATE = {
  collections: null,
};

const shopReducer = (state = INITIAL_STATE, action) => {
  switch (action.type) {
    case ShopActionTypes.UPDATE_COLLECTIONS:
      const something = {
        ...state,
        collections: action.payload,
      };
      return something;
    default:
      return state;
  }
};

export default shopReducer;