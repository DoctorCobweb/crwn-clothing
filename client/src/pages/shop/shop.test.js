import React from 'react';
import { shallow, mount } from 'enzyme';
import { combineReducers, createStore } from 'redux';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';

import { ShopPage } from './shop.component';

// because ShopPage uses hooks we need to change the way we test it.
// when using enzyme, we need to render the component using 'mount', instead of 'shallow'.
// this requires us to setup all of the infrastructure associated with fully
// rendering and mounting the component in the DOM.
// ShopPage uses the store, so we have to create a mock store, and stuff required for redux
// to function properly ie. a mock reducer, mock state.
// nb. the app also uses a persistor for storing cart data in localStorage, so we have to
// fake a persistor for the mockStore also.
//
// checkout index.js to see how the below mounting mirrows what index.js renders.
//
// i dont understand all this 100%, just following along with course material at this stage.

export const createMockStore = ({ state, reducers }) => {
  const store = createStore(combineReducers(reducers), state);
  return {
    ...store,
    persistor: {
      persist: () => null,
    }
  };
};

describe('ShopPage component', () => {
  let wrapper;
  let store;
  let mockFetchCollectionsStart;

  beforeEach(() => {
    const mockReducer = (state = { isFetching: true }, action) => state;
    const mockState = { shop: { isFetching: true } };
    mockFetchCollectionsStart = jest.fn();

    store = createMockStore({
      state: mockState,
      reducers: { shop: mockReducer },
    });

    const mockMatch = {
      path: '',
    };

    const mockProps = {
      match: mockMatch,
      fetchCollectionsStart: mockFetchCollectionsStart,
    };
    wrapper = mount(
      <BrowserRouter>
        <Provider store={store}>
          <ShopPage {...mockProps} />
        </Provider>
      </BrowserRouter>
    );
  })

  it('should render ShopPage', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('should call fetchCollectionsStart on render', () => {
    expect(mockFetchCollectionsStart).toHaveBeenCalled();
  });

});