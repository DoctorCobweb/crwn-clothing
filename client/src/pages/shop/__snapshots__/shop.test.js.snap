// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShopPage component should render ShopPage 1`] = `
"<BrowserRouter>
  <Router history={{...}}>
    <Provider store={{...}}>
      <ShopPage match={{...}} fetchCollectionsStart={[Function: mockConstructor]}>
        <styled.div>
          <StyledComponent forwardedComponent={{...}} forwardedRef={{...}}>
            <div className=\\"sc-htpNat daVAfG\\">
              <Suspense fallback={{...}}>
                <Route exact={true} path=\\"\\" component={{...}} />
                <Route path=\\"/:collectionId\\" component={{...}} />
                <Spinner>
                  <styled.div>
                    <StyledComponent forwardedComponent={{...}} forwardedRef={{...}}>
                      <div className=\\"sc-bdVaJa itPEQX\\">
                        <styled.div>
                          <StyledComponent forwardedComponent={{...}} forwardedRef={{...}}>
                            <div className=\\"sc-bwzfXH cdCsLX\\" />
                          </StyledComponent>
                        </styled.div>
                      </div>
                    </StyledComponent>
                  </styled.div>
                </Spinner>
              </Suspense>
            </div>
          </StyledComponent>
        </styled.div>
      </ShopPage>
    </Provider>
  </Router>
</BrowserRouter>"
`;
